tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,236 @@
+            'colout': 'Only for images',
+            'cutmix': 'Only for images',
+            'cutout': 'Only for images',            
+            'mixup': 'Only defined for images',
+            'alibi': 'Not compatible with simple model.',
+            'seq_length_warmup': 'Not compatible with simple model.',
+            'randaugment': 'Requires PIL dataset to test.',
+            'augmix': 'Required PIL dataset to test.',
+            'stochastic_depth': 'Only applies to ResNets.',
+            'no_op_model': 'Not compatible with this model.'
+        }
+
+resnet_skiplist = {
+            'alibi': 'Not compatible with simple model.',
+            'seq_length_warmup': 'Not compatible with simple model.',
+            'randaugment': 'Requires PIL dataset to test.',
+            'augmix': 'Required PIL dataset to test.',
+            'no_op_model': 'Not compatible with this model.'
+        }
+
+lm_skiplist = {
+            'blurpool': 'Only for CNNs',
+            'channels_last': 'Only for CNNs',
+            'colout': 'Only for images',
+            'cutmix': 'Only for images',
+            'cutout': 'Only for images',
+            'squeeze_excite': 'Not defined for language model transformers',
+            'mixup': 'Only defined for images',
+            'randaugment': 'Requires PIL dataset to test.',
+            'augmix': 'Required PIL dataset to test.',
+            'progressive_resizing': 'Only defined for images',
+            'stochastic_depth': 'Only applies to ResNets.',
+            'no_op_model': 'Not compatible with this model.'
+        }
+
+
+@pytest.mark.timeout(180)
+@pytest.mark.parametrize("world_size", [
+    pytest.param(1),
+    pytest.param(2, marks=pytest.mark.world_size(2)),
+])
+@pytest.mark.parametrize("device_hparams,deepspeed_enabled,zero_stage", [
+    pytest.param(CPUDeviceHparams(), False, None, id="cpu-ddp"),
+    pytest.param(GPUDeviceHparams(), False, None, id="gpu-ddp", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 0, id="deepspeed-zero0", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 1, id="deepspeed-zero1", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 2, id="deepspeed-zero2", marks=pytest.mark.gpu),
+])
+@pytest.mark.parametrize(
+    "seed,save_interval,save_name_format,resume_file,final_checkpoint",
+    [
+        [None, "1ep", "ep{epoch}", "ep3", "latest-rank{rank}"],  # test randomized seed saving and symlinking
+        [42, "1ep", "ep{epoch}", "ep3", "ep5"],  # test save at epoch end
+        [42, "1ep", "ep{epoch}.tgz", "ep3.tgz", "ep5.tgz"],  # test tarball with compression
+        [42, "2ba", "ba{batch}", "ba12", "ba25"],  # test save batch in partial epoch
+        [42, "1ba", "ba{batch}", "ba15", "ba25"],  # test save final batch in epoch
+        [42, "2ba", "ba{batch}", "ba16", "ba25"],  # test save batch after complete epoch
+    ],
+)
+@pytest.mark.parametrize("model_name", [None, "resnet50_synthetic", "gpt2_52m"])
```

Comment:
`model_name=None` is a bit unclear, perhaps do `default` instead? The logic below should still work.